components/tableOfContent/tableOfContent.tsx

```diff
@@ -7,15 +11,17 @@ type Props = {
 }
 
 export function TableOfContent({ toc }: Props): JSX.Element {
+  const { setVisibility } = React.useContext(TocVisibilityContext)
+
   return (
-    <div className={styles.tableOfContent}>
+    <InView as="div" className={styles.tableOfContent} onChange={(inView) => setVisibility(inView)}>
```

Comment:
It's a good practice to reduce logic inside render/return as much as possible.  Can we create a different function inside `TableOfContent` and move that logic there ?

Reply:
On point! 💥 Thanks for that 🙇🏻 

Fixed 👍🏻

Reply:
On point! 💥 Thanks for that! 🙇🏻